src/main/java/org/sonar/ucfg/BasicBlock.java

```diff
@@ -45,9 +45,9 @@ public void updateSuccs(Set<BasicBlock> successors) {
     }
   };
 
-  BasicBlock(Label label, List<Instruction.AssignCall> calls, Instruction.Terminator terminator, @Nullable LocationInFile loc) {
+  BasicBlock(Label label, List<Instruction> instructions, Instruction.Terminator terminator, @Nullable LocationInFile loc) {
     this.label = label;
-    this.calls = calls;
+    this.instructions = instructions;
```

Comment:
I know that using the `BlockBuilder` we can only add NewObject or AssignCall. But if this constructor is called from elsewhere, I think we should validate the content of the `instructions` list here, to guarantee that there will be only `Instruction.AssignCall` or `Instruction.NewObject`.

Reply:
As discussed this should be part of a refactoring outside of this PR. see SONARUCFG-12